File: MusicEventManagementSystem/MusicEventManagementSystem.Core/Models/Entities/EventOrganization/Event.cs
using MusicEventManagementSystem.Core.Enums.EventOrganization;
using MusicEventManagementSystem.Core.Models.Entities.TicketSales;

namespace MusicEventManagementSystem.Core.Models.Entities.EventOrganization
{
    public class Event
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime EventInterval { get; set; }
        public EventStatus Status { get; set; } 
        public Guid CreatedById { get; set; }
        public int LocationId { get; set; }
        //public Location Location { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }

        // Navigation property - Event - (1,N) -> TicketType
        public ICollection<TicketType> TicketTypes { get; set; } = new List<TicketType>();

        // Navigation property - Event - (1,N) -> PricingRule
        public ICollection<PricingRule> PricingRules { get; set; } = new List<PricingRule>();
    }

}

File: MusicEventManagementSystem/MusicEventManagementSystem.Client/src/frontend/event-organization/types/api/event.ts
import { EventStatus } from '../enums/EventOrganization';

export interface EventResponse {
  id: number;
  name: string;
  description: string;
  startDate: Date;
  endDate: Date;
  status: EventStatus;
  createdById: string;
  locationId: number;
  createdAt: Date;
  updatedAt: Date;
  deletedAt?: Date;
  venueIds?: number[];
  ticketTypeIds?: number[];
  pricingRuleIds?: number[];
}